src/checkpoints/checkpoints.cpp

```diff
@@ -166,6 +166,29 @@ namespace cryptonote
         block.major_version < network_version_12_checkpointing)
       return;
 
+    if (m_nettype == MAINNET && height == HF_VERSION_12_CHECKPOINTING_SOFT_FORK_HEIGHT)
+    {
+      uint64_t start_height = 0;
+      get_newest_hardcoded_checkpoint(m_nettype, &start_height);
+      start_height += 1; // Don't start deleting from the hardcoded height
+
+      for (uint64_t delete_height = start_height + (start_height % service_nodes::CHECKPOINT_INTERVAL);
+           delete_height <= height;
+           delete_height += service_nodes::CHECKPOINT_INTERVAL)
+      {
+        try
+        {
+          m_db->remove_block_checkpoint(delete_height);
```

Comment:
Why do these checkpoints need to be removed?  Can't the soft fork just begin enforcing votes & actual checkpoint reorg there, rather than restarting checkpointing there?

Reply:
Cruft we don't want/care about, maybe even buggy invalid checkpoint data, its our chance to remove it so why not?